crates/cairo-lint-core/src/lints/redundant_op.rs

```diff
@@ -0,0 +1,81 @@
+use crate::context::{CairoLintKind, Lint};
+use crate::lints::function_trait_name_from_fn_id;
+use crate::queries::{get_all_function_bodies, get_all_function_calls};
+use cairo_lang_defs::ids::ModuleItemId;
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_semantic::db::SemanticGroup;
+use cairo_lang_semantic::{Arenas, Expr, ExprFunctionCall, ExprFunctionCallArg};
+use cairo_lang_syntax::node::TypedStablePtr;
+use num_bigint::BigInt;
+
+pub struct RedundantOperation;
+
+impl Lint for RedundantOperation {
+    fn allowed_name(&self) -> &'static str {
+        "redundant_op"
+    }
+
+    fn diagnostic_message(&self) -> &'static str {
+        "This operation doesn't change the value and can be simplified."
+    }
+
+    fn kind(&self) -> crate::context::CairoLintKind {
+        CairoLintKind::RedundantOperation
+    }
+}
+
+fn is_zero(arg: &ExprFunctionCallArg, arenas: &Arenas) -> bool {
+    matches!(
+        arg,
+        ExprFunctionCallArg::Value(expr) if matches!(&arenas.exprs[*expr], Expr::Literal(val) if val.value == BigInt::from(0))
+    )
+}
```

Comment:
Maybe it's better to move `crates/cairo-lint-core/src/lints/erasing_op.rs` `is_zero` function to `crates/cairo-lint-core/src/helper.rs`, so it can be used in multiple places, so we don't define 2 different functions for the same check.